packages/web-forms/src/components/FormHeader.vue

```diff
@@ -1,20 +1,115 @@
 <script setup lang="ts">
-import Card from 'primevue/card';
-defineProps<{title: string}>();
+import { type FormLanguage, type RootNode, type SyntheticDefaultLanguage } from '@getodk/xforms-engine';
+import PrimeButton from 'primevue/button';
+import PrimeCard from 'primevue/card';
+import PrimeMenu from 'primevue/menu';
+import { ref } from 'vue';
+import FormLanguageDialog from './FormLanguageDialog.vue';
+import FormLanguageMenu from './FormLanguageMenu.vue';
+
+const props = defineProps<{form: RootNode}>();
+const languageDialogState = ref(false);
+const menu = ref<PrimeMenu>();
+
+const isFormLanguage = (lang: FormLanguage | SyntheticDefaultLanguage) : lang is FormLanguage => {
+	return !lang.isSyntheticDefault;
+}
+
+const languages = props.form.languages.filter(isFormLanguage);
+
+const print = () => window.print();
+
+const items = ref([
+	{
+		label: 'Print',
+		icon: 'icon-local_printshop',
+		command: print
+	}
+]);
+
+if(languages.length > 0){
+	items.value.unshift({
+		label: 'Change language',
+		icon: 'icon-language',
+		command: () => languageDialogState.value = true
+	})
+}
+
+const handleLanguageChange = (event: FormLanguage) => {
+	props.form.setLanguage(event);
+};
 </script>
+
 <template>
-	<Card class="form-title">
+	<!-- for desktop -->
+	<div class="hidden lg:flex justify-content-end flex-wrap gap-3 larger-screens">
+		<PrimeButton class="print-button" severity="secondary" rounded icon="icon-local_printshop" @click="print" />
+		<FormLanguageMenu 
+			:active-language="form.currentState.activeLanguage" 
+			:languages="languages" 
+			@update:active-language="handleLanguageChange"
+		/>
+	</div>
+	<PrimeCard class="form-title hidden lg:block">
 		<template #content>
-			<h1>{{ title }}</h1>
+			<!-- TODO/q: should the title be on the definition or definition.form be accessible instead of definition.bind.form -->
+			<h1>{{ form.definition.bind.form.title }}</h1>
 			<!-- last saved timestamp -->
 		</template>
-	</Card>
+	</PrimeCard>
+
+	<!-- for mobile and tablet -->
+	<div class="flex lg:hidden align-items-center smaller-screens">
```

Comment:
I was curious to see how wrapping looks for forms with long titles. Pretty good! I think we may want to iterate later on...

- More consistent alignment of controls (icons in larger views, hamburger menu in smaller). The vertical centering makes those a moving target. I think we'll probably want some design guidance on that.

- Maybe slightly reducing text size/weight over a certain character length? (Honestly the weight feels pretty heavy regardless of length, wrapping.)

Reply:
Good points, I am quoting these in #139